src/v/redpanda/admin/api-doc/transform.json

```diff
@@ -0,0 +1,162 @@
+{
+  "apiVersion": "0.0.1",
+  "swaggerVersion": "1.2",
+  "basePath": "/v1",
+  "apis": [
+    {
+      "path": "/v1/transform/deploy",
+      "operations": [
+        {
+          "method": "POST",
+          "summary": "Deploy a transform based on the provided metadata and wasm binary in the request payload. The payload should be the transform metadata in JSON format immediately followed by the WebAssembly binary without any delimiters.",
+          "nickname": "deploy_transform",
+          "consumes": [
+            "application/json+wasm"
+          ],
+          "type": "void",
+          "produces": [
+            "application/json"
+          ]
+        }
+      ]
+    },
+    {
+      "path": "/v1/transform/{name}",
+      "operations": [
+        {
+          "method": "DELETE",
```

Comment:
@rockwotj i was just joking in the design document. having a proper rest interface is extremely nice, but i thought your approach to allowing context in the body to help with future proofing was a practical approach for a big and fast moving project.

up to you!

Reply:
#its-for-roy

This is better - I'll not let my ideological objections to rest make this API worse. Looking forward to a gRPC interface in my dreams 😄